Content.Shared/GPS/Systems/HandheldGpsSystem.cs

```diff
@@ -0,0 +1,53 @@
+using Content.Shared.GPS.Components;
+using Content.Shared.Examine;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.GPS.Systems;
+
+public sealed class HandheldGpsSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<HandheldGPSComponent, ExaminedEvent>(OnExamine);
+    }
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var currentTime = _gameTiming.CurTime;
+
+        var query = EntityQueryEnumerator<HandheldGPSComponent>();
+
+        while (query.MoveNext(out var uid, out var gps))
+        {
+            if (Paused(uid) || gps.Deleted)
+                continue;
+
+            if (currentTime >= gps.NextCoordUpdate)
+            {
+                var xform = Transform(uid);
+
+                if (xform.GridUid is not { } gridUid)
+                    continue;
+
+                var pos = _transform.GetMapCoordinates(uid, xform);
+                var x = (int)pos.X;
+                var y = (int)pos.Y;
+                var posText = $"({x}, {y})";
+
+                gps.StoredCoords = Loc.GetString("handheld-gps-coordinates-title", ("coordinates", posText));
+
+                gps.NextCoordUpdate = currentTime + gps.UpdateRate;
```

Comment:
This shouldn't be done in `Update()`; this all could very reasonably be done as part of `OnExamine()` instead

Reply:
This is initially how I had it set up. The issue is that doing it on `OnExamine` means there's desync between the `StatusControl` and the Examine text because there has to be a 1.5s delay between coordinate updates.

From a networking perspective, the client would need to send out a request every 1.5s to ensure that the `StatusControl` is up to date, as the user may not have Examined it to update the `StoredCoords`.

If you forego the networking, the StatusControl would update every 1.5s via FrameUpdate like in the upstream, but the OnExamine text would be on its own interval, allowing you to get more frequent coordinate updates than 1 every 1.5s

Would it be better if the Update only queried and looped through the GPS's every 0.5s or some similar interval to reduce the per-tick performance cost?

Reply:
At the moment server is doing this when it does not need to be and it's eating up performance on nothing.

If anything this should be on the control, client already fully knows where it is so server shouldn't be involved at any point.

Reply:
@metalgearsloth Examines are always on Shared/Server in the codebase, as they're usually used to provide snippets on server-authoritative information.

The attached video shows what having Examines on the Client alone looks like:
https://github.com/user-attachments/assets/6d0ab9cb-c520-4ce7-b3c6-3d115747eaac

Reply:
Spoke with PJB and Nemo about the issue in the development chat in the Discord, gonna go with just letting the Examine bypass the cooldown on coordinate readouts.